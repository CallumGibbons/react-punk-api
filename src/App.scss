.App {
    display: flex;
    height: 100vh;
}

.Nav {
    width: 20%;
    background-color: #f0f0f0;
    padding: 20px;
    box-sizing: border-box;
    overflow-y: auto;
    border-right: 1px solid #ccc;
}

.beers {
    display: flex;
    flex-direction: column;
    align-items: center;
    flex: 1;
    padding: 20px;
    box-sizing: border-box;
    overflow-y: auto;
}
.pagination {
    display: flex;
    justify-content: center;
    margin-top: 20px;
  
    button {
      background-color: #007bff;
      color: #fff;
      border: none;
      padding: 10px 15px;
      margin: 0 5px;
      border-radius: 5px;
      cursor: pointer;
      transition: background-color 0.3s ease;
  
      &:hover {
        background-color: #0056b3;
      }
  
      &.Back-Button::after {
        content: "<";
      }
  
      &.Forward-Button::after {
        content: ">";
      }
    }
  }