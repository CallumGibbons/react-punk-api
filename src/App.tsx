import { useState } from "react";
import { BrowserRouter as Router, Routes, Route } from "react-router-dom";
import BeerProfile from "./Components/BeerProfile/BeerProfile";
import Home from "./Components/Home/Home";
import { Beer } from "./assets/Data/types";
import "./App.css";
import BeerList from "./Components/BeerList/BeerList";
import SearchBar from "./Components/Nav/SearchBar/SearchBar";
import beers from "./assets/Data/beers";

function App() {
    const [searchTerm, setSearchTerm] = useState<string>("");
    const [showBeerList, setShowBeerList] = useState<boolean>(true);

    const handleSearchChange = (value: string) => {
        setSearchTerm(value);
    };

    const filteredBeers = (beers: Beer[], searchTerm: string): Beer[] => {
        return beers.filter((beer) =>
            beer.name.toLowerCase().includes(searchTerm.toLowerCase())
        );
    };

    const handleBeerProfileDisplay = (display: boolean) => {
        setShowBeerList(display);
    };

    return (
        <Router>
            <div className="App">
                {showBeerList && (
                    <div>
                        <SearchBar searchTerm={searchTerm} onSearchChange={handleSearchChange} />
                        <BeerList beers={filteredBeers(beers, searchTerm)} />
                    </div>
                )}
            </div>
            <Routes>
                <Route path="/Home" element={<Home />} />
                <Route
                    path="/beer/:id"
                    element={<BeerProfile handleDisplay={handleBeerProfileDisplay} />}
                />
            </Routes>
        </Router>
    );
}

export default App;