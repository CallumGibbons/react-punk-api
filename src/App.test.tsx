import { render, screen, fireEvent } from "@testing-library/react";
import App from "./App";

test("renders app component", () => {
  const { container } = render(<App />);
  expect(container).toMatchSnapshot();
});

it("should render the Nav section"),
  () => {
    //1st, Arrange
    render(<App />);
    //2nd, Act
    const searchBox = screen.getByPlaceholderText(/Search beers.../i);
    const acidicFilter = screen.getByLabelText(/Acidic (Below pH4)/i);
    const abvFilter = screen.getByLabelText(/High ABV (Above 6%)/i);
    const classicFilter = screen.getByLabelText(/Classic Range/i);
    //3rd, Assert
    expect(searchBox).toBeTruthy();
    expect(abvFilter).toBeTruthy();
    expect(classicFilter).toBeTruthy();
    expect(acidicFilter).toBeTruthy();
  };

it("should not render the error message on load", () => {
  //1st, Arrange
  render(<App />);
  //2nd, Act
  const errorMessage = screen.queryByText(/Error fetching beers:/i);
  //3rd, Assert
  expect(errorMessage).toBeFalsy();
});

it(" should render beer cards on load", async () => {
  render(<App />);

  // Wait for the beer cards to be rendered
  const beerCards = await screen.findAllByTestId("beer-card");

  // Ensure that at least one beer card is rendered
  expect(beerCards.length).toBeGreaterThan(0);
});

it("should activate high abv filter on click", async () => {
  render(<App />);

  // Check if high ABV filter checkbox is initially unchecked
  const highABVFilterCheckbox = screen.getByTestId(
    "high-abv-filter"
  ) as HTMLInputElement;
  expect(highABVFilterCheckbox.checked).toBe(false);

  // Click high ABV filter
  fireEvent.click(highABVFilterCheckbox);

  // Filter is now active
  expect(highABVFilterCheckbox.checked).toBe(true);
});

it("should activate acidic filter on click", async () => {
  render(<App />);

  // Check if acidic filter checkbox is initially unchecked
  const acidicFilterCheckbox = screen.getByTestId(
    "acidic-filter"
  ) as HTMLInputElement;
  expect(acidicFilterCheckbox.checked).toBe(false);

  // Click acidic filter
  fireEvent.click(acidicFilterCheckbox);

  // Filter is now active
  expect(acidicFilterCheckbox.checked).toBe(true);
});

it("should activate classic filter on click", async () => {
  render(<App />);

  // Check if classic filter checkbox is initially unchecked
  const classicFilterCheckbox = screen.getByTestId(
    "classic-filter"
  ) as HTMLInputElement;
  expect(classicFilterCheckbox.checked).toBe(false);

  // Click classic filter
  fireEvent.click(classicFilterCheckbox);

  // Filter is now active
  expect(classicFilterCheckbox.checked).toBe(true);
});

it("should render 20 beer cards on load", async () => {
  render(<App />);

  const initialBeerCards = await screen.findAllByTestId("beer-card");
  const initialBeerCount = initialBeerCards.length;

  expect(initialBeerCount).toEqual(20);
});

it("should render beer profile when name is clicked", async () => {
  render(<App />);
  const profileLink = await screen.findByText("Buzz");
  fireEvent.click(profileLink);
  const profile = screen.getByTestId("beer-profile") as HTMLDivElement;
  expect(profile).toBe(true);
});
