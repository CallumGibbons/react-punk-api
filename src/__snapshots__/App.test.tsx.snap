// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders app component 1`] = `
<div>
  <div
    class="App"
  >
    <div
      class="nav"
    >
      <input
        class="search-bar"
        placeholder="Search beers..."
        type="text"
        value=""
      />
      <div
        class="filter-box-container"
      >
        <input
          class="filter-box"
          data-testid="high-abv-filter"
          type="checkbox"
        />
        <label
          class="filter-box-text"
        >
          High ABV (Above 6%)
        </label>
      </div>
      <div
        class="filter-box-container"
      >
        <input
          data-testid="acidic-filter"
          type="checkbox"
        />
        <label
          class="filter-box-text"
        >
          Acidic (Below pH4)
        </label>
      </div>
      <div
        class="filter-box-container"
      >
        <input
          class="filter-box"
          data-testid="classic-filter"
          type="checkbox"
        />
        <label
          class="filter-box-text"
        >
          Classic Range 
        </label>
      </div>
    </div>
    <div />
    <div
      class="beers"
    >
      <div
        class="beer-list"
      />
      <div
        class="pagination"
      >
        <button
          class="Back-Button"
        />
        <button
          class="Return-Button"
        >
          Return
        </button>
        <button
          class="Forward-Button"
        />
      </div>
    </div>
  </div>
</div>
`;
